clients/slo/slo.go

```diff
@@ -0,0 +1,220 @@
+// @license
+// Copyright 2025 Dynatrace LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slo
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+)
+
+const endpointPath = "platform/slo/v1/slos"
+
+func NewClient(client *rest.Client) *Client {
+	c := &Client{
+		restClient: client,
+	}
+	return c
+}
+
+type Client struct {
+	restClient *rest.Client
+}
+
+func (c *Client) List(ctx context.Context) (api.PagedListResponse, error) {
+	var retVal api.PagedListResponse
+
+	for hasNextPage, nextPageKey := true, ""; hasNextPage; {
+		var listResponse api.ListResponse
+		var err error
+
+		nextPageKey, listResponse, err = listPage(ctx, c, nextPageKey)
+		if err != nil {
+			return nil, fmt.Errorf(errMsg, "list", err)
+		}
+
+		retVal = append(retVal, listResponse)
+		hasNextPage = nextPageKey != ""
+	}
+
+	return retVal, nil
+
+}
+
+func listPage(ctx context.Context, c *Client, pageKey string) (string, api.ListResponse, error) {
+	ro := rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests}
+	if pageKey != "" {
+		ro.QueryParams = url.Values{"page-key": {pageKey}}
+	}
+
+	resp, err := c.restClient.GET(ctx, endpointPath, ro)
+	if err != nil {
+		return "", api.ListResponse{}, fmt.Errorf(errMsg, "list", err)
+	}
+
+	return processListResponse(resp)
+}
+
+func (c *Client) Get(ctx context.Context, id string) (api.Response, error) {
+	if id == "" {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "get", id, errors.New(`argument "id" is empty`))
+	}
+
+	path, err := url.JoinPath(endpointPath, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "get", id, err)
+	}
+
+	resp, err := c.restClient.GET(ctx, path, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
```

Comment:
Follow up: Please add tests if the network breaks, so when GET, POST, etc return an error. 
Imo these are important error cases to test.